src/server/mod.rs

```diff
@@ -112,8 +119,12 @@ impl<'a, H: Handler + 'static> Server<'a, H, HttpListener> {
     /// Binds to a socket, and starts handling connections using a task pool.
     pub fn listen_threads<T: ToSocketAddrs>(self, addr: T, threads: usize) -> ::Result<Listening> {
         let listener = try!(match self.ssl {
+            #[cfg(feature = "openssl")]
             Some(SslConfig::CertAndKey(cert, key)) => HttpListener::https(addr, cert, key),
+            #[cfg(feature = "openssl")]
             Some(SslConfig::Context(ssl_context)) => HttpListener::https_with_context(addr, ssl_context),
+            #[cfg(not(feature = "openssl"))]
+            Some(_) => panic!("Internal error - we have an SSL config even though SSL is disabled"),
```

Comment:
Instead of this panic, can the cfg just be put on the `ssl: Option` property of the `Server`?